analyzers/src/SonarAnalyzer.Common/Rules/CommentsShouldNotBeEmptyBase.cs

```diff
@@ -0,0 +1,126 @@
+﻿/*
+ * SonarAnalyzer for .NET
+ * Copyright (C) 2015-2023 SonarSource SA
+ * mailto: contact AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using Microsoft.CodeAnalysis.Text;
+
+namespace SonarAnalyzer.Rules;
+
+public abstract class CommentsShouldNotBeEmptyBase<TSyntaxKind> : SonarDiagnosticAnalyzer<TSyntaxKind>
+    where TSyntaxKind : struct
+{
+    private const string DiagnosticId = "S4663";
+
+    protected abstract string GetCommentText(SyntaxTrivia trivia);
+    protected abstract bool IsValidTriviaType(SyntaxTrivia trivia);
```

Comment:
Can be implemented with the language facade.
```suggestion
    protected bool IsCommentTrivia(SyntaxTrivia trivia) => ...
```

Reply:
Follow-up to a previous comment:

Should I check the `RawKind` on the base class?
I do not have a `.Kind()` method, since they are different extension methods for c# and vb.

Reply:
Yes. RawKind with a cast to `TSyntaxKind` if needed.

Reply:
Same as the other similar comments, I can't seem to cast `int` to `TSyntaxKind` or the other way around.